starlette_admin/helpers.py

```diff
@@ -151,3 +152,34 @@ def not_none(value: Optional[T]) -> T:
     if value is not None:
         return value
     raise ValueError("Value can not be None")  # pragma: no cover
+
+
+def timedelta_to_components(td: timedelta):
+    # Constants
+    seconds_in_minute = 60
+    seconds_in_hour = 3600
+    seconds_in_week = 604800
+
+    # Total seconds in the timedelta
+    total_seconds = td.total_seconds()
+
+    # Calculate each component
+    weeks = int(total_seconds // seconds_in_week)
+    total_seconds -= weeks * seconds_in_week
+
+    days = td.days % 7
+    hours = td.seconds // seconds_in_hour
+    minutes = (td.seconds % seconds_in_hour) // seconds_in_minute
+    seconds = td.seconds % seconds_in_minute
+    milliseconds = td.microseconds // 1000
+    microseconds = td.microseconds % 1000
+
+    return {
```

Comment:
I would prefer using [format_timedelta](https://babel.pocoo.org/en/latest/dates.html#time-delta-formatting) from babel) and extend `StringField` to skip the display and render template.

Reply:
There's a loss of precision with that though, for example (delta is 1 week and 5 hours):

```
>>> delta
datetime.timedelta(days=7, seconds=18000)
>>> format_timedelta(delta, locale='en_US', granularity='hour')
'1 week'
```

It all depends on the use case of course, but for me I'm making some cron job stuff, and 5 hours make a lot of difference.. if internally the value is precise, the displayed version should reflect that in my opinion.

What do you think? I could return a `str` from the helper function and skip the template if needed.

Reply:
Found this: https://humanize.readthedocs.io/en/stable/time/#humanize.time.precisedelta

Would that work?

Reply:
There's something I'm confused about. The `serialize_value` returns a value that the form can pull existing data from, right? Is this method also used for creating a string representation of the data, such as with babel or humanize?

If that's the case, then I don't know how to read the info from `data` in the form, as it is now a string...

Reply:
The benefit of [format_timedelta](https://babel.pocoo.org/en/latest/dates.html#time-delta-formatting) is the support for internationalization. The way humanize handles internationalization will lead to race conditions when handling multiple requests.

One way is to find all the components and call `format_timedelta` on each of them and then join with a separator(space) 

> If that's the case, then I don't know how to read the info from data in the form, as it is now a string...

(you probably already figured it out) you can return different values based on the request action `request.state.action`